guides/puppet-web-apps/html-template-guidelines/README.md

```diff
@@ -0,0 +1,157 @@
+        <!-- layout goes here -->
+    </template>
+</template>
+```
+
+A more in-depth explanation of `starcounter-composition` and why functionality and layout should be separated can be found in the [article linked above](https://starcounter.io/layout-compositions-html-partials/).
+
+### Guideline 2: Define the Functional Elements
+
+Functional elements are elements that either interact with the user visually, such as `<h1>`, `<p>` and `<span>` elements, or provides some kind of interaction between the user and the application, such as `<input>` and `<button>` elements. To start building the HTML template, these elements should be added to the `dom-bind` template as seen in Guideline 1.
+
+To have complete clarity in what counts as a functional elements, here are some examples:
+
+{% raw %}
+
+**Simple elements:**
+
+* `<button value="{{model.Submit::click}}">Submit</button>`
+* `<h1>Very Important Headline</h1>`
+* `<a href="/here/there">There</a>`
+* `<starcounter-include partial="{{model.MyPartial}}"></starcounter-include>`
+
+**Nested elements:**
+
+There are two kinds of nested elements that count as functional elements:
+
+1. Elements such as `<table>`, `<nav>`, and `<ul>` which are not just pure container elements but also provide some kind of functionality to it's children.
+```html
+<table>
+      <tr>
+          <th>Name</th>
+          <th>Age</th>
+      </tr>
+      <tr>
+          <td>Alice</td>
+          <td>42</td>
+      </tr>
+</table>
+```
+2. All elements, including `<div>` elements, when the child is a `<template>`.
+```html
+<div>
+      <template is="dom-if" if="{{model.Person.Alive}}">
+          <strong>Still going strong</strong>
+      </template>
+</div>
+```
+
+All these element should be put on the root level of the `dom-bind` template.
+
+### Guideline 3: Add Slots to Functional Elements
+
+To give clearer semantic meaning to functional elements when mixing with other application, explicit slot names are used. This is how elements look when blending without explicit slot names: `<content select="[slot='MyApp/0']"></content>`, the zero is simply the index of the element in the template. With an explicit slot name, it becomes much clearer what kind of element it is: `<content select="[slot='MyApp/MainHeadline']"></content>`.
+
+Slot names are added as attributes to the functional elements like so:
+
+```html
+<button slot="MyApp/SubmitButton" value="{{model.Submit::click}}">Submit</button>
+
+<div slot="MyApp/LifeReport">
+    <template is="dom-if" if="{{model.Person.Alive}}">
+        <strong>Still going strong</strong>
+    </template>
+</div>
+```
+
+### Guideline 4: Create Layout in `starcounter-composition`
+
+As outlined in guideline 1, the layout of the HTML template should be included within the `<template is="starcounter-composition">`.
+
+The elements with a slot are grabbed into the `starcounter-composition` using the following syntax: `<content select="[slot='AppName/ElementName']"></content>`.

```

Comment:
That's not what happens. None of elements is grabbed into `starcounter-composition`.

If you want to give a syntax for distributing functional elements in Shadow DOM composition, I'd rather say:

 To distribute functional elements within the Shadow DOM layout, you should use regular `<slot>`/`<content>` element:
```html
<slot name="AppName/ElementName"></slot>
<content select="[slot='AppName/ElementName']"></content>
```
> If you are not familiar with Shadow DOM, take a look at a great [article by Eric Bidelman](https://developers.google.com/web/fundamentals/getting-started/primers/shadowdom), or [the spec](http://w3c.github.io/webcomponents/spec/shadow/)